### PR TITLE
Introduced await as a more feature rich version of eventually for whe…

### PR DESCRIPTION
**Pull request by sksamuel**
…n you want to use custom polling periods #793

### CODE REVIEW

**Comment by LeoColman** [comment]
Did you bring these guys straight from Awaitility? Some of them?

File: kotlintest-assertions/src/main/kotlin/io/kotlintest/eventually/Eventually.kt [hunk 1]
```diff
@@ -1,5 +1,8 @@
-package io.kotlintest
+package io.kotlintest.eventually
```
Comment (LeoColman) [inline kotlintest-assertions/src/main/kotlin/io/kotlintest/eventually/Eventually.kt]: Moving this guy to another package breaks client's code. 

If it breaks already, we don't even need to deprecate, I suppose

File: kotlintest-assertions/src/main/kotlin/io/kotlintest/eventually/FibonacciPollInterval.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * Create an instance of the [FibonacciPollInterval] with a supplied time unit.
+ *
+ * @param offset   The fibonacci offset. Eg if offset is 5 and poll count is 1 then the duration will be base * fib(6).
+ * @param base The duration that is multiplied by the fibonacci value
+ */
+class FibonacciPollInterval(private val base: Duration, private val offset: Int) : PollInterval {
+
+  init {
+    if (offset <= -1) {
```
Comment (LeoColman) [inline kotlintest-assertions/src/main/kotlin/io/kotlintest/eventually/FibonacciPollInterval.kt]: require(offset >0) { "message" }

File: kotlintest-assertions/src/main/kotlin/io/kotlintest/eventually/PollInterval.kt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+/**
+ * A [PollInterval] represents how often Awaitility will pause before reevaluating the supplied condition.
+ * <p/>
+ * Note that the name "poll interval" is a bit misleading. It's actually a delay between two successive condition evaluations.
+ * I.e if the condition evaluation takes 5 ms and a fixed poll interval of 100 ms is used then the next condition evaluation will happen at
+ * (approximately) 105 ms. It's called <tt>PollInterval</tt> for historic reasons.
+ *
+ * @since 1.7.0
+ */
+interface PollInterval {
```
Comment (LeoColman) [inline kotlintest-assertions/src/main/kotlin/io/kotlintest/eventually/PollInterval.kt]: We need to improve docs on this class

**Comment by LeoColman** [comment]
I think that await might not be the best name here.

It is very used in CoRoutines

File: kotlintest-assertions/src/main/kotlin/io/kotlintest/eventually/Eventually.kt (hunk 1 above)
Comment (sksamuel) [reply kotlintest-assertions/src/main/kotlin/io/kotlintest/eventually/Eventually.kt]: I'll add a deprecated alias.

File: kotlintest-assertions/src/main/kotlin/io/kotlintest/eventually/FibonacciPollInterval.kt (hunk 2 above)
Comment (sksamuel) [reply kotlintest-assertions/src/main/kotlin/io/kotlintest/eventually/FibonacciPollInterval.kt]: :+1:

**Comment by sksamuel** [comment]
await is only used in coroutines as part of deferred, not as a standalone function, so it won't clash. I could call it `until`.

**Comment by sksamuel** [comment]
I've renamed it `until`, added better (not copied) docs, and moved eventually back to the original package.

Please merge if you're happy.

**Pull request merged**